docs/samples/Microsoft.ML.Samples/Static/LightGBMMulticlassWithInMemoryData.cs

```diff
@@ -0,0 +1,163 @@
+using Microsoft.ML.StaticPipe;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Samples.Static
+{
+    class LightGBMMulticlassWithInMemoryData
+    {
+        /// <summary>
+        /// Number of features per example used in <see cref="MultiClassLightGbmStaticPipelineWithInMemoryData"/>.
+        /// </summary>
+        private const int _featureVectorLength = 10;
+
+        /// <summary>
+        /// Data point used in <see cref="MultiClassLightGbmStaticPipelineWithInMemoryData"/>. A data set there 
+        /// is a collection of <see cref="NativeExample"/>.
+        /// </summary>
+        private class NativeExample
+        {
+            [VectorType(_featureVectorLength)]
+            public float[] Features;
+            [ColumnName("Label")]
+            public string Label;
+            public uint LabelIndex;
+            public uint PredictedLabelIndex;
+            [VectorType(4)]
+            // The probabilities of being "AA", "BB", "CC", and "DD".
+            public float[] Scores;
+
+            public NativeExample()
+            {
+                Features = new float[_featureVectorLength];
+            }
+        }
+
+        /// <summary>
+        /// Helper function used to generate <see cref="NativeExample"/>s.
+        /// </summary>
+        private static List<NativeExample> GenerateRandomExamples(int count)
+        {
+            var examples = new List<NativeExample>();
+            var rnd = new Random(0);
+            for (int i = 0; i < count; ++i)
+            {
+                var example = new NativeExample();
+                var res = i % 4;
+                // Generate random float feature values.
+                for (int j = 0; j < _featureVectorLength; ++j)
+                {
+                    var value = (float)rnd.NextDouble() + res * 0.2f;
+                    example.Features[j] = value;
+                }
+
+                // Generate label based on feature sum.
+                if (res == 0)
+                    example.Label = "AA";
+                else if (res == 1)
+                    example.Label = "BB";
+                else if (res == 2)
+                    example.Label = "CC";
+                else
+                    example.Label = "DD";
+
+                // The following three attributes are just placeholder for storing prediction results.
+                example.LabelIndex = default;
+                example.PredictedLabelIndex = default;
+                example.Scores = new float[4];
+
+                examples.Add(example);
+            }
+            return examples;
+        }
+
+        public void MultiClassLightGbmStaticPipelineWithInMemoryData()
+        {
+            // Create a general context for ML.NET operations. It can be used for exception tracking and logging,
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 1, conc: 1);
```

Comment:
>seed: 1, conc: 1 [](start = 42, length = 16)

let's omit those in samples, and go with defaults as much as we can.

Reply:
Sure.

---
In reply to: [244875689](https://github.com/dotnet/machinelearning/pull/1953#discussion_r244875689) [](ancestors = 244875689)